From e888accbfa60724afb2ba08ec9710fc5f94bcd10 Mon Sep 17 00:00:00 2001
From: Gyuho Lee <leegyuho@amazon.com>
Date: Tue, 28 Sep 2021 19:38:47 +0000
Subject: [PATCH] --EKS-PATCH-- apiserver/healthz: upper log verbosity for
 "kms-provider-0"

To silence

    cannot exclude some health checks, no health checks are installed
    matching "kms-provider-0"

This is logged when external health checker calls
"/healthz?exclude=kms-provider-0" against API server
that does not enable KMS encryption. Reduce such logs
to minimize the noise and log billing.

Signed-off-by: Gyuho Lee <leegyuho@amazon.com>
---
 .../src/k8s.io/apiserver/pkg/server/healthz/healthz.go    | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go b/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
index e80b8501edf..75735565f2a 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
@@ -235,8 +235,12 @@ func handleRootHealth(name string, checks ...HealthChecker) http.HandlerFunc {
 			}
 		}
 		if excluded.Len() > 0 {
-			fmt.Fprintf(&individualCheckOutput, "warn: some health checks cannot be excluded: no matches for %s\n", formatQuoted(excluded.List()...))
-			klog.Warningf("cannot exclude some health checks, no health checks are installed matching %s",
+			// EKS-PATCH: to exclude "kms-provider-0" user-induced error, EKS has a separate alarm for encryption provider
+			verbosity := klog.V(6)
+			if verbosity.Enabled() {
+				fmt.Fprintf(&individualCheckOutput, "warn: some health checks cannot be excluded: no matches for %s\n", formatQuoted(excluded.List()...))
+			}
+			verbosity.Infof("cannot exclude some health checks, no health checks are installed matching %s",
 				formatQuoted(excluded.List()...))
 		}
 		// always be verbose on failure
-- 
2.32.0

